pkg/operator/certrotation/metadata.go

```diff
@@ -0,0 +1,37 @@
+package certrotation
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func ensureMetadataUpdate(secret *corev1.Secret, owner *metav1.OwnerReference, additionalAnnotations AdditionalAnnotations) bool {
+
+	// Existing secret not found - no need to update metadata (will be done by needNewSigningCertKeyPair / NeedNewTargetCertKeyPair)
+	if len(secret.ResourceVersion) == 0 {
+		return false
+	}
```

Comment:
this probably belongs to the other function, otherwise nothing sets the ownerRef - or does it?

Reply:
Yeah, good point, this also shows different function flows

Reply:
Fixed